packages/embed/src/utils/create-iframe/get-form-event-handler.ts

```diff
@@ -1,6 +1,12 @@
-import { ActionableOptions } from '../../base'
+import { ActionableOptions, UrlOptions } from '../../base'
+import { setupGoogleAnalyticsInstance } from '../'
 
-export const getFormReadyHandler = (embedId: string, options: ActionableOptions) => {
+export interface FormReadyOptionTypes extends ActionableOptions, UrlOptions {}
+
+export const getFormReadyHandler = (embedId: string, iframe: HTMLIFrameElement, options: FormReadyOptionTypes) => {
+  if (options.shareGoogleAnalyticsInstance) {
+    setupGoogleAnalyticsInstance(iframe, embedId)
```

Comment:
please do not call it here. It would be hard to guess that `getFormReadyHandler` performs this side effect of setting up the GA instance.

Reply:
🤔  I see but this needs to be called when the iframe is fully loaded, hence on `form-ready` (and the `setupGoogleAnalyticsInstance ` needs the iframe passed to it). Where would it be the best place to load it?

Reply:
I think I would add another listener to the `form-ready` event with proper name.

Reply:
But where do we want to add this event? We agreed not in `create-iframe.ts`, we agreed not here ☝️

Reply:
The previous version was using React so it was loaded in `componentDidMount` for each embed